addons/sale/tests/test_sale_order.py

```diff
@@ -25,6 +30,22 @@ def setUpClass(cls):
         cls.setUpClassicProducts()
         cls.setUpAccountJournal()
 
+        # Create a pricelist with/without discount policy
+        cls.pricelist_with_disct = cls.Pricelist.create({
+            'name': 'Pricelist A',
+            'discount_policy': 'with_discount',
+        })
+        cls.pricelist_without_disct = cls.Pricelist.create({
+            'name': 'Pricelist B',
+            'discount_policy': 'without_discount',
+        })
+
+        # Create the SO with pricelist
+        cls.sale_order_pricelist = cls.env['sale.order'].create({
```

Comment:
Why don't you reuse the SO without pricelist, and add it in the pricelist tests ?

Reply:
For the pricelist's onchange effect on so lines, first we've to select pricelist of SO, then by making SO lines we got the effect of onchange (i.e on One2many fields). If we've SO with 4 SO lines and then select pricelist of SO, we'll not get onchange effect on that SO line that's why we create a SO without SO lines in setup and inside pricelist test method we've applied first pricelist of SO and creating SO line to get result of onchange.